src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -199,98 +207,83 @@ private class TaskLauncherActor(
       logger.debug(s"After delay update $status")
 
     case msg @ RateLimiterActor.DelayUpdate(spec, delayUntil) if spec != runSpec =>
-      logger.warn(s"Received delay update for other runSpec: $msg")
+      logger.warn(s"Received delay update for other run spec ${spec.id} version ${spec.version} and delay $delayUntil. Current run spec is ${runSpec.id} version ${runSpec.version}")
 
     case RecheckIfBackOffUntilReached => OfferMatcherRegistration.manageOfferMatcherStatus()
   }
 
   private[this] def receiveTaskLaunchNotification: Receive = {
-    case InstanceOpSourceDelegate.InstanceOpRejected(op, reason) if inFlight(op) =>
-      removeInstance(op.instanceId)
-      logger.debug(s"Task op '${op.getClass.getSimpleName}' for ${op.instanceId} was REJECTED, reason '$reason', rescheduling. $status")
-
-      op match {
-        // only increment for launch ops, not for reservations:
-        case _: InstanceOp.LaunchTask => instancesToLaunch += 1
-        case _: InstanceOp.LaunchTaskGroup => instancesToLaunch += 1
-        case _ => ()
-      }
-
-      OfferMatcherRegistration.manageOfferMatcherStatus()
-
     case InstanceOpSourceDelegate.InstanceOpRejected(op, TaskLauncherActor.OfferOperationRejectedTimeoutReason) =>
-      // This is a message that we scheduled in this actor.
-      // When we receive a launch confirmation or rejection, we cancel this timer but
-      // there is still a race and we might send ourselves the message nevertheless, so we just
-      // ignore it here.
-      logger.debug(s"Ignoring task launch rejected for '${op.instanceId}' as the task is not in flight anymore")
+      // Reschedule instance with provision timeout..
+      if (inFlightInstanceOperations.exists(_.instanceId == op.instanceId)) {
+        instanceMap.get(op.instanceId).foreach { instance =>
+          import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
What do you think about introducing a dedicated ec for this purpose?

Reply:
I am not sure it makes sense. This piece of code is executed so rarely if ever...